packages/synchronize/lib/src/sources.dart

```diff
@@ -0,0 +1,36 @@
+part of '../synchronize.dart';
+
+/// The source the sync uses to sync from and to.
+abstract class SyncSource<T1, T2> {
```

Comment:
```suggestion
abstract interface class SyncSource<T1, T2> {
```

Reply:
Not possible because interface classes can't be extended outside the package

Reply:
Ah yes I've run into this before :(

I wonder how it's possible to extend dart interfaces.
```dart
class CustomUri implements Uri{}
```

is valid code :thinking:

Reply:
Maybe it's just allowed for the sdk?

Reply:
RTFM 🤦‍♀️
https://dart.dev/language/class-modifiers#interface

We can't extend it but we can implement it.
Makes sense xD